apps/sap-commerce-cloud/frontend/src/components/Dialog/Dialog.tsx

```diff
@@ -29,269 +29,220 @@ interface DialogProps {
   sdk: DialogAppSDK<AppParameters>;
 }
 
-interface State {
-  baseSite: string;
-  baseSites: string[];
-  query: string;
-  page: number;
-  totalPages: number;
-  products: Product[];
-  selectedProducts: string[];
-  errors: Error[];
-}
-
-export default class Dialog extends React.Component<DialogProps, State> {
-  state: State = {
-    baseSite: '',
-    baseSites: [],
-    query: '',
-    page: 0,
-    totalPages: 0,
-    products: [],
-    selectedProducts: [],
-    errors: [],
-  };
-
-  componentDidMount() {
-    this.loadBaseSites().then(() => {
-      this.load();
-    });
-  }
-
-  load = async () => {
+const Dialog: React.FC<DialogProps> = ({ sdk }) => {
+  const [baseSite, setBaseSite] = useState('');
+  const [baseSites, setBaseSites] = useState<string[]>([]);
+  const [query, setQuery] = useState('');
+  const [page, setPage] = useState(0);
+  const [totalPages, setTotalPages] = useState(0);
+  const [products, setProducts] = useState<Product[]>([]);
+  const [selectedProducts, setSelectedProducts] = useState<string[]>([]);
+  const [errors, setErrors] = useState<ErrorType[]>([]);
+
+  useEffect(() => {
+    const initialize = async () => {
+      await loadBaseSites();
+      load();
+    };
+    initialize();
+  }, []);
+
+  const load = async () => {
```

Comment:
this might be a good candidate for useCallback if we make it a dep of useEffect

Reply:
good idea!